iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/DanglingLineAdderImpl.java

```diff
@@ -12,9 +12,64 @@
 /**
  *
  * @author Geoffroy Jamgotchian <geoffroy.jamgotchian at rte-france.com>
+ * @author Anne Tilloy <anne.tilloy at rte-france.com>
+ *
  */
 class DanglingLineAdderImpl extends AbstractInjectionAdder<DanglingLineAdderImpl> implements DanglingLineAdder {
 
+    class GenerationAdderImpl implements GenerationAdder {
+
+        double minP;
+        double maxP;
+        double targetP;
+        double targetQ;
+        boolean voltageRegulationOn;
+        double targetV;
+
+        @Override
+        public GenerationAdder setTargetP(double targetP) {
+            this.targetP = targetP;
+            return this;
+        }
+
+        @Override
+        public GenerationAdder setMaxP(double maxP) {
+            this.maxP = maxP;
+            return this;
+        }
+
+        @Override
+        public GenerationAdder setMinP(double minP) {
+            this.minP = minP;
+            return this;
+        }
+
+        @Override
+        public GenerationAdder setTargetQ(double targetQ) {
+            this.targetQ = targetQ;
+            return this;
+        }
+
+        @Override
+        public GenerationAdder setVoltageRegulationOn(boolean voltageRegulationOn) {
+            this.voltageRegulationOn = voltageRegulationOn;
+            return this;
+        }
+
+        @Override
+        public GenerationAdder setTargetV(double targetV) {
+            this.targetV = targetV;
+            return this;
+        }
+
+        @Override
+        public DanglingLineAdder add() {
+            ValidationUtil.checkActivePowerLimits(DanglingLineAdderImpl.this, minP, maxP);
```

Comment:
In GeneratorAdderImpl, we check a lot of thing I would expect to find here such as:
 - ValidationUtil.checkMinP(this, minP);
- ValidationUtil.checkMaxP(this, maxP);
- ValidationUtil.checkActivePowerSetpoint(this, targetP);
- ValidationUtil.checkVoltageControl(this, voltageRegulatorOn, targetV, targetQ);

Reply:
Yes but in this case minP/maxP are not mandatory (which leads to no check), same goes for targets.

Reply:
I am not sure how generation data is actually used in CGMES/UCTE, @annetill do you have any input on this?

Reply:
Indeed from CGMES and from UCTE, we will create minimal generation part of a dangling line:
- Sometimes a generation that has only targetQ and targetP ;
- Sometimes a generation that has only targetP, targetV and voltageRegulationOn.